platform/security/secure-boot/src/test/groovy/org/codice/ddf/security/boot/SecureBootSpec.groovy

```diff
@@ -0,0 +1,258 @@
+        def userHomeSysProp = userHome.getRoot().toString()
+        def ddfHomeSysProp = userHome.newFolder("ddf").toString()
+        def systemService = Mock(SystemService)
+        System.properties.'ddf.home' = ddfHomeSysProp
+        System.properties.'user.home' = userHomeSysProp
+
+        when:
+        def secureBoot = new SecureBoot(systemService) {
+            boolean securityManagerEnabled() {
+                return null;
+            }
+            void systemExit(Exception e) {
+                exitedDueToSystemServiceFailureToExit = true
+            }
+        }
+        secureBoot.init()
+
+        then:
+        0 * systemService.halt("0")
+        exitedDueToSystemServiceFailureToExit == false
+    }
+
+    def 'System boots when security manager disabled and ddf installed outside of user home dir'() {
+        setup:
+        def exitedDueToSystemServiceFailureToExit = false
+        def userHomeSysProp = userHome.newFolder("tomPenny").toString()
+        def ddfHomeSysProp = ddfHome.newFolder("ddf").toString()
+        def systemService = Mock(SystemService)
+        System.properties.'ddf.home' = ddfHomeSysProp
+        System.properties.'user.home' = userHomeSysProp
+
+        when:
+        def secureBoot = new SecureBoot(systemService) {
+            boolean securityManagerEnabled() {
+                return null;
+            }
+            void systemExit(Exception e) {
+                exitedDueToSystemServiceFailureToExit = true
+            }
+        }
+        secureBoot.init()
+
+        then:
+        0 * systemService.halt("0")
+        exitedDueToSystemServiceFailureToExit == false
+    }
+
+    def 'SystemService throws exception when asked to halt'() {
+        setup:
+        def exitedDueToSystemServiceFailureToExit = false
+        def userHomeSysProp = userHome.getRoot().toString()
+        def ddfHomeSysProp = userHome.newFolder("ddf").toString()
+        def systemService = Mock(SystemService)
+        def securityManager = Mock(SecurityManager)
+        System.properties.'ddf.home' = ddfHomeSysProp
+        System.properties.'user.home' = userHomeSysProp
+
+        when:
+        def secureBoot = new SecureBoot(systemService) {
+            boolean securityManagerEnabled() {
+                return securityManager;
+            }
+
+            void systemExit(Exception e) {
+                exitedDueToSystemServiceFailureToExit = true
+            }
+        }
+        secureBoot.init()
+
+        then:
+        1 * systemService.halt("0") >> { throw new Exception("unable to halt")}
+        exitedDueToSystemServiceFailureToExit == true
+    }
+
+    def 'System exits normally when attempting to read invalid ddf.home'() {
+        setup:
+        def exitedDueToSystemServiceFailureToExit = false
+        def userHomeSysProp = userHome.getRoot().toString()
+        def ddfHomeSysProp = "/" + System.currentTimeMillis() + "/" + UUID.randomUUID() + "/ddf"
```

Comment:
❓will that work on Windows?

Reply:
I have tested these on Windows and they work.